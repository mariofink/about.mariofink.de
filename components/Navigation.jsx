import Link from "next/link";

export default function Navigation() {
  return (
    <nav className="container mx-auto px-6 mt-6 max-w-prose">
      <ul>
        <li>
          <Link href="/">
            <a>About Me</a>
          </Link>
        </li>
        <li>
          <Link href="/cv">
            <a>CV/Resume</a>
          </Link>
        </li>
      </ul>
    </nav>
  );
}
