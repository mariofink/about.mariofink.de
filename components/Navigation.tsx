import NavItem from "./NavItem";

const Navigation = () => (
  <nav className="flex justify-center sm:gap-6 sm:px-6 mt-3">
    <NavItem caption="About Me" href="/" />
    <NavItem caption="CV/Résumé" href="/cv/" />
    <NavItem caption="Tools &amp; Tech" href="/tech/" />
  </nav>
);

export default Navigation;
