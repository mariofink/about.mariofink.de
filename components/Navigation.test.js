import React from "react";
import renderer from "react-test-renderer";
import Navigation from "./Navigation";

/**
 * Needed because the navigation renders NavItems,
 * which use the Link component, which require the next/router
 */
jest.mock("next/dist/client/router", () => require("next-router-mock"));

describe("Navigation", () => {
  it("renders", () => {
    const tree = renderer.create(<Navigation />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
