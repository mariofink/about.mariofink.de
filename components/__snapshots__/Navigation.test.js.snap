// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation renders 1`] = `
<nav
  className="flex justify-center sm:gap-6 sm:px-6 mt-3"
>
  <a
    className="py-4 px-2 font-semibold text-green-500"
    href="/"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    About Me
  </a>
  <a
    className="py-4 px-2 font-semibold text-green-500"
    href="/cv"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    CV/Resume
  </a>
  <a
    className="py-4 px-2 font-semibold text-green-500"
    href="/tech"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    Tools & Tech
  </a>
</nav>
`;
